File: weibo.cn/downloader/downloader.go
package main

import (
	"crypto/md5"
	"crypto/rand"
	"encoding/base64"
	"encoding/hex"
	"fmt"
	"github.com/PuerkitoBio/goquery"
	"io"
	"net/http"
	"os"
)

func getMd5String(s string) string {
	h := md5.New()
	h.Write([]byte(s))
	return hex.EncodeToString(h.Sum(nil))

}

func getGuid() string {
	b := make([]byte, 48)

	if _, err := io.ReadFull(rand.Reader, b); err != nil {
		return ""
	}
	return getMd5String(base64.URLEncoding.EncodeToString(b))
}

func main() {
	for i := 0; i < 1000; i++ {

		doc, err := goquery.NewDocument("http://login.weibo.cn/login/")
		if err != nil {
			continue
		}
		doc.Find("img").Each(func(i int, contentSelection *goquery.Selection) {
			if i == 0 {
				imageURL, exist := contentSelection.Attr("src")
				if exist {
					fmt.Println(imageURL)
					resp, err := http.Get(imageURL)
					if err != nil {
						return
					}
					fmt.Println(resp.ContentLength)
					if resp.ContentLength != -1 {
						return
					}

					filename := "./captchas/" + getGuid() + ".gif"
					file, err := os.Create(filename)
					if err != nil {
						return
					}
					io.Copy(file, resp.Body)
					resp.Body.Close()
					fmt.Println(filename)
				}

			}
		})
	}

}

File: weibo.cn/downloader/downloader.py
# coding:utf-8
import requests
import uuid
from PIL import Image
import os
from bs4 import BeautifulSoup

url = "http://login.weibo.cn/login/"
for i in range(2000):
    try:
        resp = requests.get(url)
        bsObj = BeautifulSoup(resp.content, "html.parser")
        image_url = str(bsObj.img['src'])
        #print(image_url)
        resp = requests.get(image_url)
        filename = str(uuid.uuid4()) + ".gif"
        with open("./captchas/" + filename, 'wb') as f:
           f.write(resp.content)

        try:
            with Image.open("./captchas/" + filename) as im:
                im.save("./captchas/" + filename.split('.gif')[0] + ".png")

        except Exception as ex:
            print(Exception, ":", ex)
        #os.remove("./captchas/" + filename)
        print(filename)
    except Exception as ex:
        print(Exception, ":", ex)
